Src/FluentAssertions/Specialized/ExecutionTimeAssertions.cs

```diff
@@ -10,18 +12,39 @@ namespace FluentAssertions.Specialized
     /// </summary>
     public class ExecutionTimeAssertions
     {
-        private readonly TimeSpan executionTimeSpan;
-
-        private readonly string actionDescription;
+        private readonly ExecutionTime execution;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="ExecutionTime"/> class.
         /// </summary>
         /// <param name="executionTime">The execution on which time must be asserted.</param>
         public ExecutionTimeAssertions(ExecutionTime executionTime)
         {
-            executionTimeSpan = executionTime.ExecutionTimeSpan;
-            actionDescription = executionTime.ActionDescription;
+            this.execution = executionTime;
+        }
+
+        /// <summary>
+        /// Checks the executing action if it satisfies a condition.
+        /// If the execution runs into an exeption, then this will rethrow it.
+        /// </summary>
+        /// <param name="condition">Condition to check on the current elapsed time.</param>
+        /// <param name="result">The expected result when polling is stopped.</param>
```

Comment:
Minor:
Maybe it's just me, but `result` indicates a computed value to me, whereas this is a stop condition.
So maybe rename to something like `stopCondition`, `expectedResult` or?